akka-http-core/src/main/java/akka/http/javadsl/model/MediaTypes.java

```diff
@@ -187,6 +187,119 @@ private MediaTypes() { }
+        akka.http.scaladsl.model.MediaType.Compressibility comp = compressible ?
+                akka.http.scaladsl.model.MediaType.Compressible$.MODULE$ : akka.http.scaladsl.model.MediaType.NotCompressible$.MODULE$;
+
+        scala.collection.immutable.List<String> fileEx =
+                scala.collection.JavaConversions.asScalaBuffer(java.util.Arrays.asList(fileExtensions)).toList();
+
+        return akka.http.scaladsl.model.MediaType.audio(subType, comp, fileEx);
+    }
+
+    public static MediaType.Binary image(String subType, boolean compressible, String... fileExtensions) {
+        akka.http.scaladsl.model.MediaType.Compressibility comp = compressible ?
+                akka.http.scaladsl.model.MediaType.Compressible$.MODULE$ : akka.http.scaladsl.model.MediaType.NotCompressible$.MODULE$;
+
+        scala.collection.immutable.List<String> fileEx =
+                scala.collection.JavaConversions.asScalaBuffer(java.util.Arrays.asList(fileExtensions)).toList();
+
+        return akka.http.scaladsl.model.MediaType.image(subType, comp, fileEx);
+    }
+
+    public static MediaType.Binary message(String subType, boolean compressible, String... fileExtensions) {
+        akka.http.scaladsl.model.MediaType.Compressibility comp = compressible ?
+                akka.http.scaladsl.model.MediaType.Compressible$.MODULE$ : akka.http.scaladsl.model.MediaType.NotCompressible$.MODULE$;
+
+        scala.collection.immutable.List<String> fileEx =
+                scala.collection.JavaConversions.asScalaBuffer(java.util.Arrays.asList(fileExtensions)).toList();
+
+        return akka.http.scaladsl.model.MediaType.message(subType, comp, fileEx);
+    }
+
+    public static MediaType.WithOpenCharset text(String subType, String... fileExtensions) {
+        scala.collection.immutable.List<String> fileEx =
+                scala.collection.JavaConversions.asScalaBuffer(java.util.Arrays.asList(fileExtensions)).toList();
+
+        return akka.http.scaladsl.model.MediaType.text(subType, fileEx);
+    }
+
+    public static MediaType.Binary video(String subType, boolean compressible, String... fileExtensions) {
+        akka.http.scaladsl.model.MediaType.Compressibility comp = compressible ?
+                akka.http.scaladsl.model.MediaType.Compressible$.MODULE$ : akka.http.scaladsl.model.MediaType.NotCompressible$.MODULE$;
+
+        scala.collection.immutable.List<String> fileEx =
+                scala.collection.JavaConversions.asScalaBuffer(java.util.Arrays.asList(fileExtensions)).toList();
+
+        return akka.http.scaladsl.model.MediaType.video(subType, comp, fileEx);
+    }
+
+    // arguments have been reordered due to varargs having to be the last argument
+    // should we create multiple overloads of this function?
+    public static MediaType.Binary customBinary(String mainType, String subType, boolean compressible, java.util.Map<String, String> params, boolean allowArbitrarySubtypes, String... fileExtensions) {
```

Comment:
Sounds good, I'll have a look if the Scala one should be changed to or no need to then